bellman/src/groth16/verifier.rs

```diff
@@ -22,9 +22,9 @@ pub fn verify_proof<'a, E: MultiMillerLoop>(
     pvk: &'a PreparedVerifyingKey<E>,
     proof: &Proof<E>,
     public_inputs: &[E::Fr],
-) -> Result<bool, SynthesisError> {
+) -> Result<(), VerificationError> {
     if (public_inputs.len() + 1) != pvk.ic.len() {
-        return Err(SynthesisError::MalformedVerifyingKey);
+        return Err(VerificationError::InvalidVerifyingKey);
```

Comment:
This will in practice cause `InvalidVerifyingKey` to be treated identically to an invalid proof by the calling code, when actually it indicates an error in how that code is using the API. This seems like a (minor) regression, although I can't see how it could cause a security failure in tested code.

Reply:
I know it doesn't affect callers in zcashd who were *already* collapsing those two cases.

Reply:
I think this is only true in practice if the calling code discards error information and propagates boolean values. I think the zcash_proofs does this. Otherwise the error is propagated upward until it is handled (by matching on the error type) or reported (which report will include the information indicating a programming error).  So I don't think there is any conflation here.